src/Nix/Builtins.hs

```diff
@@ -706,9 +719,11 @@ genericClosure = fromValue @(AttrSet (NThunk m)) >=> \s ->
 
 replaceStrings :: MonadNix e m => m (NValue m) -> m (NValue m) -> m (NValue m) -> m (NValue m)
 replaceStrings tfrom tto ts =
-    fromNix tfrom >>= \(from :: [Text]) ->
-    fromNix tto   >>= \(to   :: [Text]) ->
-    fromValue ts  >>= \(s    :: Text) -> do
+    fromNix tfrom >>= \(nsFrom :: [NixString]) ->
+    fromNix tto   >>= \(nsTo   :: [NixString]) ->
+    fromValue ts  >>= \(ns    :: NixString) -> do
+        let from = map principledStringIgnoreContext nsFrom
+        let to = map principledStringIgnoreContext nsTo
```

Comment:
I think we want to just leave `to` as a list of `NixString`, we want to include the relevant context if we actually do a replacement. See https://github.com/NixOS/nix/blob/2f8255cba1597aa1d7afefcac0a33bec63e0ceb6/src/libexpr/primops.cc#L1988-L1994 and https://github.com/NixOS/nix/blob/2f8255cba1597aa1d7afefcac0a33bec63e0ceb6/src/libexpr/primops.cc#L2016-L2017